piptools/utils.py

```diff
@@ -152,89 +169,33 @@ def flat_map(fn, collection):
     return chain.from_iterable(map(fn, collection))
 
 
-def lookup_table(values, key=None, keyval=None, unique=False, use_lists=False):
+def lookup_table(
+    values: Iterable[Union[_VT, Tuple[_KT, _VT]]],
+    key: Optional[Callable[[_VT], _KT]] = None,
+) -> Dict[_KT, Set[_VT]]:
     """
     Builds a dict-based lookup table (index) elegantly.
-
-    Supports building normal and unique lookup tables.  For example:
-
-    >>> assert lookup_table(
-    ...     ['foo', 'bar', 'baz', 'qux', 'quux'], lambda s: s[0]) == {
-    ...     'b': {'bar', 'baz'},
-    ...     'f': {'foo'},
-    ...     'q': {'quux', 'qux'}
-    ... }
-
-    For key functions that uniquely identify values, set unique=True:
-
-    >>> assert lookup_table(
-    ...     ['foo', 'bar', 'baz', 'qux', 'quux'], lambda s: s[0],
-    ...     unique=True) == {
-    ...     'b': 'baz',
-    ...     'f': 'foo',
-    ...     'q': 'quux'
-    ... }
-
-    For the values represented as lists, set use_lists=True:
-
-    >>> assert lookup_table(
-    ...     ['foo', 'bar', 'baz', 'qux', 'quux'], lambda s: s[0],
-    ...     use_lists=True) == {
-    ...     'b': ['bar', 'baz'],
-    ...     'f': ['foo'],
-    ...     'q': ['qux', 'quux']
-    ... }
-
-    The values of the resulting lookup table will be lists, not sets.
-
-    For extra power, you can even change the values while building up the LUT.
-    To do so, use the `keyval` function instead of the `key` arg:
-
-    >>> assert lookup_table(
-    ...     ['foo', 'bar', 'baz', 'qux', 'quux'],
-    ...     keyval=lambda s: (s[0], s[1:])) == {
-    ...     'b': {'ar', 'az'},
-    ...     'f': {'oo'},
-    ...     'q': {'uux', 'ux'}
-    ... }
-
     """
```

Comment:
Here's how to improve this user-input check:
```suggestion
    """
    if key is None and values and any(not isinstance(v, tuple) for v in values):
        raise ValueError("The `key` function must be specified when the `values` are not empty.")

```

Reply:
Addressed in 1c125852462698c1d40e50f796f81c73f3411f2b.